docs/DeveloperGuide.md

```diff
@@ -1,29 +1,204 @@
 
+**Architecture of Main Components** <br/>
+`DinerDirector` entry point to the application is DinerDirector. Initially, `DinerDirector` class interact with `UI` class to 
+prompt user for inputs. Once user keys in the input text, the input will be redirected to `Parser` class. `Parser` class will manage the error 
+handling of user's input. Next, any logic will be executed by `Command` object. If there exists any interaction with `Entity`, `Command` will 
+request to `Manager` to assist the CRUD operations.  
+
+### Command Component (Zheng Rong)
+
+##### The command component consists of the following:
+
+![Command_Package_Diagram](uml/images/CommandPackageDiagram.png)
+
+- 4 subcomponents: Command, HelpCommand, ExitCommand, IncorrectCommand
+
+- 4 sub packages with the following names: deadline, meeting, menu and staff
+    - Each package contains the 3 general commands:
+        - AddCommand
+        - DeleteCommand
+        - ViewCommand
+    - The general commands have specific names depending on the package it is in.
+
+The command component consist of the Abstract `Command` class.
+It has multiple child `Command` classes that inherits from the Abstract `Command` Class.
+Each child `Command` class accepts arguments that are parsed from the `Parser` class and set those arguments to private variables in their respective classes, if there is any private variables to be set.
+An `execute` method will execute the operations needed for each individual commands.
+An `isExit` method will return a boolean value that decides whether the program should exit after this command is called.
+
+
+### Manager Component (PeiHao)
+The manager component consists of four different managers, 
+in which the list of entity is initialized and the methods implementing the entity are written inside.
+- DeadlineManager: This class contains an ArrayList of deadlines and methods implementing the deadlines like addDeadline, printDeadline and deleteDeadline.
+- DishManager: This class contains an ArrayList of dishes and methods like addDish, deleteDish, viewDish and 
+stringOfDish which returns the dish information.
+- MeetingManager: This class contains an ArrayList of meetings and methods like addMeeting, printMeetings and deleteMeeting.
+- StaffManager: This class contains an ArrayList of staffs and methods like addStaff, deleteStaffByName, findStaffByName and getStaffString which print all the staffs' information.
+
+### Ui Component (HuiQi)
+The Ui Component consists of the TextUi class that handles interactions between the app and the user.
+
+The `TextUi`class performs the following functions:
+1. Reading the user input (Empty inputs will be ignored)
+
+    method: `getUserInput()`
+2. Printing messages generated by the app to the terminal
+
+    method: `printBanner()`, `printMessage()`
+
+### Utils Component (Darren)
+
+![UtilsParserClassDiagram](uml/images/UtilsParserClassDiagram.png)    
```

Comment:
The (C) should be removed from your diagrams, and you should use `+`/`-`/`#`.

This goes for your other class diagrams too.